app/controllers/events_controller.rb

```diff
@@ -180,8 +180,27 @@ def typed_group_events_path(group, event_type, options = {})
   end
 
   def permitted_attrs
-    attrs = entry.class.used_attributes
-    attrs + self.class.permitted_attrs
+    remove_unused_nested_attributes(
+      entry.class.used_attributes +
+      self.class.permitted_attrs
+    )
```

Comment:
Das Problem aus Vogelperspektive:
- Derselbe `events_controller` wird für eine ganze Klassen-Hierarchie von Events, Kursen, Lagern, etc. verwendet.
- Je nach dem mit welchem Model der Controller gerade zu tun hat, muss er hier eine andere Liste von erlaubten Attributen angeben. Gibt er zu wenige an, können die fehlenden Felder nicht mehr via Formular geändert werden. Gibt er zu viele an, insbesondere Attribute von `nested` Entities (das Problem in hitobito/hitobito_insieme#92), dann schlägt das Speichern ganz fehl, weil das Model diese überflüssigen nested Entities nicht kennt und nicht akzeptiert.
- Die angegebene Liste wird hier zusammengesetzt aus der `used_attributes` Angabe vom aktuellen Model, kombiniert mit pauschalen `permitted_attrs` Attributen vom Controller selber.
- Der Insieme Wagon definiert die Kurs-spezifischen `nested_attributes` namens `course_record_attributes` in den pauschalen `permitted_attrs` des Controllers.

Die hier implementierte Lösung filtert die kombinierte Liste nachträglich, basierend auf der tatsächlichen Klasse des Models. Das ist eher kompliziert, weil z.B. `group_ids` eine Ausnahme ist, die dann doch wieder allen Events-Subklassen erlaubt ist. Es ist so für Wagons auch fast unmöglich, neue pauschale `nested_attributes` zu ergänzen, wenn diese mal wirklich nötig wären.

Besser fände ich es, wenn wir die `course_record_attributes` der Insieme verschieben: Aus den pauschalen `permitted_attrs` des Controllers entfernen, und stattdessen nur in denjenigen Models irgendwie deklarieren, die auch wirklich nested `course_record_attributes` brauchen. Dann muss man nicht mehr nachträglich filtern, was jetzt tatsächlich erlaubt ist.

Auf dem Model sind die `used_attributes` bisher überall wo ich nachgeschaut habe eine flache Liste von Symbols, also nicht direkt geeignet um nested Attributes in Hashes da reinzupacken. Wir könnten aber entweder ein neues `class_attribute :used_nested_attrs` einführen, oder die nested Attributes basierend auf `nested_attributes_options` berechnen.

Reply:
@kronn falls du meiner Analyse und dem Vorschlag nicht zustimmst, kannst du trotzdem mergen. Der Code an sich ist gut, ich stelle nur das Lösungskonzept in Frage.
(aber sobald jemand mal irgendwo `nested_attributes` mit `group_ids` hinzufügt wirds wieder kaputt gehen)

Reply:
Ich habe es noch so umgeschrieben, dass es grundsätzlich nur nested attributes betrachtet und alles andere durchwinkt. Deinen Einwand habe ich jetzt als Kommentar aufgenommen. Für jetzt finde ich die Lösung gut genug, bei der nächsten Anpassung hat man dann mehr Bedarf und Wissen, um es anders zu lösen.